src/utils/map.ts

```diff
@@ -7,12 +7,27 @@ export module Utils {
    * https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map
    */
   export class Map<K, V> {
-    private _keyValuePairs: { key: K; value: V; }[] = [];
+    private _keyValuePairs: { key: K; value: V; }[];
+    private _es6Map: any;
+
+    public constructor() {
+      if (typeof (<any>window).Map === "function") {
+        this._es6Map = new (<any>window).Map();
+      } else {
+        this._keyValuePairs = [];
+      }
+    }
 
     public set(key: K, value: V) {
       if (Utils.Math.isNaN(key)) {
         throw new Error("NaN may not be used as a key to the Map");
       }
+
+      if (this._es6Map != null) {
+        this._es6Map.set(key, value);
+        return this;
+      }
```

Comment:
However, if we leave the line there, the two maps will behave the same, and there will be no differences between browsers (IE failing in unexpected ways)